### ISSUE TITLE
Generated project's package.json dependencies are behind David-DM versions

### ISSUE THREAD

**Issue by EliKrumholz** [angular-fullstack/generator-angular-fullstack#1931]
Item | Version
----- | -----
generator-angular-fullstack | 3.7.4
Node | 4.4.5
npm | 3.9.3
Operating System | OS X 10

Item | Answer
----- | -----
Transpiler | Babel
Markup | HTML
CSS | SCSS
Router | ui-router
Build Tool | Grunt
Client Tests | Jasmine
DB | MongoDB
Auth | Y

The dependencies in the package.json file of my generated project are behind the versions listed on the [David-DM page](https://david-dm.org/angular-fullstack/angular-fullstack-deps#info=dependencies&view=table).

This leads to a bunch of issues, most notably grunt-node-inspector runs into errors when I run `npm install` - this particular issue has popped up here a number of times so I won't rehash it (updating the version in package.json file seems to fix it) - **but why are all the dependencies behind the David-DM versions in the first place, am I doing something wrong?**

package.json file:
```
{
  "name": "test-project",
  "version": "0.0.0",
  "main": "server/app.js",
  "dependencies": {
    "express": "~4.9.0",
    "morgan": "~1.0.0",
    "body-parser": "~1.5.0",
    "method-override": "~1.0.0",
    "serve-favicon": "~2.0.1",
    "cookie-parser": "~1.0.1",
    "express-session": "~1.0.2",
    "errorhandler": "~1.0.0",
    "compression": "~1.0.1",
    "lodash": "~2.4.1",
    "ejs": "~0.8.4",
    "mongoose": "~4.0.3",
    "jsonwebtoken": "^5.0.0",
    "express-jwt": "^3.0.0",
    "passport": "~0.2.0",
    "passport-local": "~0.1.6",
    "passport-facebook": "latest",
    "passport-twitter": "latest",
    "passport-google-oauth": "latest",
    "composable-middleware": "^0.3.0",
    "connect-mongo": "^0.8.1",
    "socket.io": "^1.0.6",
    "socket.io-client": "^1.0.6",
    "socketio-jwt": "^3.0.0"
  },
  "devDependencies": {
    "grunt": "~0.4.4",
    "grunt-autoprefixer": "~0.7.2",
    "grunt-wiredep": "~1.8.0",
    "grunt-concurrent": "~0.5.0",
    "grunt-contrib-clean": "~0.5.0",
    "grunt-contrib-concat": "~0.4.0",
    "grunt-contrib-copy": "~0.5.0",
    "grunt-contrib-cssmin": "~0.9.0",
    "grunt-contrib-htmlmin": "~0.2.0",
    "grunt-contrib-imagemin": "~0.7.1",
    "grunt-contrib-jshint": "~0.10.0",
    "grunt-contrib-uglify": "~0.4.0",
    "grunt-contrib-watch": "~0.6.1",
    "karma-babel-preprocessor": "^5.2.1",
    "grunt-babel": "~5.0.0",
    "grunt-google-cdn": "~0.4.0",
    "grunt-newer": "~0.7.0",
    "grunt-ng-annotate": "^0.2.3",
    "grunt-rev": "~0.1.0",
    "grunt-svgmin": "~0.4.0",
    "grunt-usemin": "~2.1.1",
    "grunt-env": "~0.4.1",
    "grunt-node-inspector": "~0.1.5",
    "grunt-nodemon": "~0.2.0",
    "grunt-angular-templates": "^0.5.4",
    "grunt-dom-munger": "^3.4.0",
    "grunt-protractor-runner": "^1.1.0",
    "grunt-injector": "~0.5.4",
    "grunt-karma": "~0.8.2",
    "grunt-build-control": "~0.4.0",
    "grunt-mocha-test": "~0.10.2",
    "grunt-contrib-sass": "^0.7.3",
    "jit-grunt": "^0.5.0",
    "time-grunt": "~0.3.1",
    "grunt-express-server": "~0.4.17",
    "grunt-open": "~0.2.3",
    "open": "~0.0.4",
    "jshint-stylish": "~0.1.5",
    "connect-livereload": "~0.4.0",
    "karma-ng-scenario": "~0.1.0",
    "karma-firefox-launcher": "~0.1.3",
    "karma-script-launcher": "~0.1.0",
    "karma-html2js-preprocessor": "~0.1.0",
    "karma-ng-jade2js-preprocessor": "^0.1.2",
    "karma-jasmine": "~0.1.5",
    "karma-chrome-launcher": "~0.1.3",
    "requirejs": "~2.1.11",
    "karma-requirejs": "~0.2.1",
    "karma-coffee-preprocessor": "~0.2.1",
    "karma-jade-preprocessor": "0.0.11",
    "karma-phantomjs-launcher": "~0.1.4",
    "karma": "~0.12.9",
    "karma-ng-html2js-preprocessor": "~0.1.0",
    "supertest": "~0.11.0",
    "should": "~3.3.1"
  },
  "engines": {
    "node": ">=0.10.0"
  },
  "scripts": {
    "start": "node server/app.js",
    "test": "grunt test",
    "update-webdriver": "node node_modules/grunt-protractor-runner/node_modules/protractor/bin/webdriver-manager update"
  },
  "private": true
}
```

**Comment by Awk34** [comment]
You must have not generated with 3.7.4. You can check the dependencies for the latest version here: https://github.com/angular-fullstack/generator-angular-fullstack/blob/3.7.4/templates/app/_package.json

**Comment by EliKrumholz** [comment]
You are correct - I was actually running the generator from an old version I had installed pre-NVM that never got properly removed. I found this stackoverflow question helpful: http://stackoverflow.com/questions/11177954/how-do-i-completely-uninstall-node-js-and-reinstall-from-beginning-mac-os-x

Once I completely uninstalled everything and reinstalled from scratch it everything looks and runs well.

Thanks!
